File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.js
const fs = require('fs');

function main() {
    let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
    let [N, M] = input.shift().split(' ').map(x => +x),
        baskets = new Array(N + 1).fill(0).map((v, i) => i);

    for (let i = 0; i < M; i++) {
        let [a, b] = input[i].split(' ').map(x => +x);

        baskets.splice(a, b - a + 1, ...baskets.slice(a, b + 1).reverse());
    }

    baskets.shift();
    console.log(...baskets);
}

main();

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.py
import sys
input = sys.stdin.readline
N,M = map(int, input().split(" "))
arr = [i+1 for i in range(N)]

for k in range(M):
    i, j = map(int, input().split(" "))
    arr1 = arr[i - 1 :j]
    arr1.reverse()
    arr[i-1:j] = arr1
for i in range(N):
    print(arr[i],end=" ")